client/components/domains/map-domain-step/index.jsx

```diff
@@ -173,13 +177,20 @@ var MapDomainStep = React.createClass( {
 	},
 
 	addMappingToCart: function( domain ) {
+		this.addPremiumPlanToCart();
 		upgradesActions.addItem( cartItems.domainMapping( { domain: domain } ) );
 
 		if ( this.isMounted() ) {
 			page( '/checkout/' + this.props.selectedSite.slug );
 		}
 	},
 
+	addPremiumPlanToCart() {
```

Comment:
This function seems [duplicated](https://github.com/Automattic/wp-calypso/pull/5000/files#diff-99f1f1d8ba147cd06cfd4a0d8d195d34R456).